adapter.go

```diff
@@ -609,7 +613,18 @@ func (a *Adapter) RemoveFilteredPolicy(sec string, ptype string, fieldIndex int,
 	if fieldIndex <= 5 && 5 < fieldIndex+len(fieldValues) {
 		line.V5 = fieldValues[5-fieldIndex]
 	}
-	err := a.rawDelete(a.db, *line)
+	err = a.rawDelete(a.db, *line)
+	return err
+}
+
+// checkQueryfield make sure the field won't be empty (string --> "")
+func checkQueryField(fieldValues []string) error {
+	var err error
+	for _, fieldValue := range fieldValues {
+		if fieldValue == "" {
+			return errors.New("the query field cannot be empty string (\"\"), please check")
```

Comment:
I would not suggest throwing error for empty field value because it can be a genuine case where I want to use empty values

Reply:
Allow some empty values, but not all empty field values, how about it ?

Reply:
I would say that we leave the entire validation upto the users. Maybe a use case for all allowing all empty fields is that someone wants to do a cleanup for all the empty records.
That way we leave flexibility in the system

Reply:
If user pass in all empty value, all ptype policies will be deleted now. This PR will prohibit this global deletion. If you want to allow this action, you can use ``e.RemoveFilteredPolicy(-1,"")``,it means delete all. Empty string value will not be used as a conditional parameter in sql ``where``, because it has no meaning.